polyply/src/load_library.py

```diff
@@ -101,12 +137,64 @@ def load_library(name, lib_names, extra_files):
     """
     force_field = vermouth.forcefield.ForceField(name)
     if lib_names and extra_files:
-        lib_files = _resolve_lib_paths(lib_names, DATA_PATH)
+        lib_files = _resolve_lib_paths(lib_names, DATA_PATH,
+                                       FORCE_FIELD_PARSERS.keys())
         all_files = lib_files + extra_files
     elif lib_names:
-        all_files = _resolve_lib_paths(lib_names, DATA_PATH)
+        all_files = _resolve_lib_paths(lib_names, DATA_PATH,
+                                       FORCE_FIELD_PARSERS.keys())
     elif extra_files:
         all_files = extra_files
 
     read_ff_from_files(all_files, force_field)
     return force_field
+
+
+def read_build_options_from_files(paths, topology):
+    """
+    read the input files for the build options and
+    molecule templates.
+
+    Parameters
+    ----------
+    paths: list[`os.path`]
+           List of valid file paths
+    topology: :class:`polyply.src.topology`
+
+    Returns
+    -------
+
+    """
+    for path in paths:
+        with open(path, 'r') as file_:
+            lines = file_.readlines()
+        read_build_file(lines, topology.molecules, topology)
```

Comment:
```suggestion
    for path in paths:
        with open(path, 'r') as file_:
            read_build_file(file_, topology.molecules, topology)
```
Don't read the entire file in memory if you can avoid it. `file_` is an iterator.